### PR TITLE
docs: Add angular-fontawesome documentation to icon

### PR DESCRIPTION
**Pull request by rapropos**
Font Awesome 5 provides [a separate `fa-icon` component in a library called angular-fontawesome](http://github.com/FortAwesome/angular-fontawesome). It was not immediately obvious to me how to use this in places where `mdc-icon` is expected (specifically, inside textfields) until I read the source and figured that content projection would work. Feel free to ignore this PR if you do not want to support arbitrary content projection inside `mdc-icon`, but I think the note might be helpful to other FA5 and angular-mdc-web users.

This PR also adds a link to the icon demo in README.md.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,6 +45,7 @@ Each component uses [material-components-web](https://github.com/material-compon
 | fab | Available | [View Demo](https://trimox.github.io/angular-mdc-web/#/fab-demo) |
 | form-field | Available | [View Demo](https://trimox.github.io/angular-mdc-web/#/form-field-demo) |
 | grid-list | Available | [View Demo](https://trimox.github.io/angular-mdc-web/#/grid-list-demo) |
+| icon | Available | [View Demo](https://trimox.github.io/angular-mdc-web/#/icon-demo) |
```
Comment (trimox) [inline README.md]: Nice catch on missing link! 👍

File: src/demo-app/components/icon-demo/icon-demo.html [hunk 2]
```diff
@@ -110,3 +110,13 @@ <h3 class="demo-content__headline">Using Font Awesome w/size modifier</h3>
     <mdc-icon #fasize fontSet="fa" fontIcon="fa-map fa-4x"></mdc-icon>
   </div>
 </div>
+
+<h2>Font Awesome 5</h2>
```
Comment (trimox) [inline src/demo-app/components/icon-demo/icon-demo.html]: @rapropos This has me thinking.. 🤔 

We should implement a service for registering a font's CSS class. 

Set a default `fontSet` class
```ts
mdcIconRegistry.defaultFontSetClass('fa');
```

Making Font Awesome usage with `mdc-icon` easy throughout app
```html
<mdc-icon>cakeIcon</mdc-icon>
```

Comment (rapropos) [reply src/demo-app/components/icon-demo/icon-demo.html]: That is one option, but at least personally I ran into several counterintuitive problems with FA5's SVG magic in an Angular environment. For example, structural directives such as `*ngIf` and dynamic binding of classes to change icons simply don't work on `<i>` elements. The `<fa-icon>` directive in angular-fontawesome smooths over all of these problems, allowing one to use FA5 in a much more idiomatic Angular way.

I considered submitting a PR using `fa-icon` under the hood in `mdc-icon`, but after thinking a bit about it, it seemed out of scope for angular-mdc-web, and once I discovered that content projection worked without modifying the existing angular-mdc-web codebase, I decided that I thought simply documenting that fact and illustrating how to use it with angular-fontawesome was the most prudent course of action.

Comment (trimox) [reply src/demo-app/components/icon-demo/icon-demo.html]: Hey, @rapropos I plan to revisit PR after finishing v0.36.0-0 update. I'll need some free time to review FA5.

Comment (trimox) [reply src/demo-app/components/icon-demo/icon-demo.html]: Turns out v0.36.0-0 required some `mdc-icon` improvements..

New public UX state methods for PR https://github.com/trimox/angular-mdc-web/pull/1021

**Comment by trimox** [comment]
Revisiting for #1175 effort.

**Comment by trimox** [comment]
@rapropos Have you had an opportunity to try out the new `mdc-icon` and `MdcIconRegistry` ? Hoping Font Awesome usability has improved for ya.

**Comment by trimox** [comment]
Closing since the upcoming documentation improvements should resolve. Thanks again for the research.

**Pull request closed without merging**